src/components/BookCollection.js

```diff
@@ -1,22 +1,32 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import Book from './Book';
+import { getBooksData } from '../redux/books/booksSlice';
 
 const BookCollection = () => {
-  const bookList = useSelector((state) => state.book.bookList);
+  const {
+    books, isLoading, error, isSuccessful,
+  } = useSelector((store) => store.book);
+  const dispatch = useDispatch();
 
-  return (
-    <>
-      {bookList.map((book) => (
-        <Book
-          key={book.item_id}
-          title={book.title}
-          author={book.author}
-          item_id={book.item_id}
-        />
-      ))}
-    </>
-  );
+  useEffect(() => {
+    dispatch(getBooksData());
+  }, [dispatch, isSuccessful]);
+
+  if (isLoading) {
+    return (
+      <div>
+        <p>Is Loading...</p>
+      </div>
+    );
+  }
+  if (error) return <h2>Something went wrong</h2>;
+  if (books.length === 0) return <h2>No books available. Please come check later</h2>;
+  return books.map((book) => (
+    <div key={book.id}>
+      <Book key={book.id} book={book} />
+    </div>
+  ));
```

Comment:
- [OPTIONAL] Currently, there seems to be an issue where this component is being re-rendered excessively upon making requests, resulting in the reloading state being displayed and obstructing the visibility of the existing list of books. To address this concern, we should focus on implementing a more efficient rendering strategy. It's imperative that we ensure our page only re-render when there are actual changes, preventing unnecessary removal and addition of components for each book modification. By doing so, we can significantly improve the user experience and create a smoother interaction within your app.


### YOUR APP BEHAVIOUR

   ![chrome-capture-2023-8-1](https://github.com/SaveryIV/Bookstore-React/assets/87197412/1e8f5f81-1b8c-4a4d-ba28-e7addb13673e)